src/analysis/retail/monk/windwalker/modules/apl/AplCheck.tsx

```diff
@@ -4,28 +4,53 @@ import RESOURCE_TYPES from 'game/RESOURCE_TYPES';
 import SpellLink from 'interface/SpellLink';
 import { suggestion } from 'parser/core/Analyzer';
 import { AnyEvent } from 'parser/core/Events';
-import aplCheck, { build, CheckResult, PlayerInfo, tenseAlt } from 'parser/shared/metrics/apl';
+import aplCheck, {
+  Apl,
+  build,
+  CheckResult,
+  Condition,
+  PlayerInfo,
+  tenseAlt,
+} from 'parser/shared/metrics/apl';
 import annotateTimeline from 'parser/shared/metrics/apl/annotate';
 import { AplRuleProps } from 'parser/shared/metrics/apl/ChecklistRule';
 import {
   and,
+  buffMissing,
   buffPresent,
+  buffRemaining,
   buffStacks,
   describe,
   hasResource,
   hasTalent,
   lastSpellCast,
   not,
+  or,
+  spellCooldownRemaining,
 } from 'parser/shared/metrics/apl/conditions';
 import {
   ASCENSION_ENERGY_MAX_ADDITION,
   BASE_ENERGY_MAX,
   INNER_PEACE_ENERGY_MAX_ADDITION,
 } from '../resources/EnergyCapTracker';
 
-const hasChi = (min: number) => hasResource(RESOURCE_TYPES.CHI, { atLeast: min });
+const hasChi = (min: number): Condition<number> => {
+  const resource = hasResource(RESOURCE_TYPES.CHI, {
+    atLeast:
+      hasTalent(TALENTS.ORDERED_ELEMENTS_TALENT) && buffPresent(TALENTS.STORM_EARTH_AND_FIRE_TALENT)
+        ? min - 1
+        : min,
+  });
```

Comment:
very similar note here, except that you can't use `combatant.hasBuff` in this location because if you do it will check the buffs present at the start of the fight rather than at each event.

you could do it without relying on a `combatant` parameter like this, i think:

```tsx
const resource = or(
  and(hasTalent(TALENTS.ORDERED_ELEMENTS_TALENT), buffPresent(TALENTS.STORM_EARTH_AND_FIRE_TALENT), hasResource(RESOURCE_TYPES.CHI, { atLeast: min - 1 })),
  hasResource(RESOURCE_TYPES.CHI, { atLeast: min })
);
```